diff --git a/setuptools/command/_requirestxt.py b/setuptools/command/_requirestxt.py
index b0c2d7059..87596119f 100644
--- a/setuptools/command/_requirestxt.py
+++ b/setuptools/command/_requirestxt.py
@@ -106,7 +106,7 @@ def _write_requirements(stream, reqs):
     def append_cr(line):
         return line + '\n'
 
-    lines = map(append_cr, lines)
+    lines = map(append_cr, sorted(lines))
     stream.writelines(lines)
 
 
